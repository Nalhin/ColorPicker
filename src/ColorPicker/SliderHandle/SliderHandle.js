import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { colorArrayToRgba } from '../Utils';
import { SELECTOR_SIZE } from '../constants';

const Wrapper = styled.div`
  position: relative;
  touch-action: none;
`;

const Handle = styled.div`
  position: absolute;
  width: ${SELECTOR_SIZE}px;
  height: 14px;
  border-radius: 50%;
  border: 2px #fff solid;
  &:hover {
    cursor: pointer;
  }
`;

const SliderHandle = ({ color, position, children, click }) => {
  return (
    <Wrapper onMouseDown={click} onTouchStart={click}>
      <Handle
        style={{
          top: `${position.y}px`,
          left: `${position.x - SELECTOR_SIZE / 2}px`,
          background: colorArrayToRgba(color),
        }}
      />
      {children}
    </Wrapper>
  );
};

SliderHandle.propTypes = {
  click: PropTypes.func.isRequired,
  color: PropTypes.arrayOf(PropTypes.number).isRequired,
  children: PropTypes.element.isRequired,
  position: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number,
  }).isRequired,
};

export default SliderHandle;
