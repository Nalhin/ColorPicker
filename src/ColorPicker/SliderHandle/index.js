import SliderHandle from './SliderHandle';

export default SliderHandle;
