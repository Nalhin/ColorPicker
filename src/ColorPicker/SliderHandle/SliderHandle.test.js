import React from 'react';
import { render } from '@testing-library/react';
import SliderHandle from './SliderHandle';

describe('Slider Handle', () => {
  // toHaveStyles doesn't work with rgba values
  const props = {
    color: [140, 150, 22, 1],
    position: { x: 40, y: 50 },
    children: <div />,
    click: () => {},
  };

  it('Should render correctly', () => {
    const { container } = render(
      <SliderHandle {...props}>
        <div />
      </SliderHandle>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  // it('Should change background color', () => {
  //   const color = 'rgb(140, 150, 22, 1)';
  //   const { container } = render(
  //     <SliderHandle {...props}>
  //       <div />
  //     </SliderHandle>,
  //   );
  //   expect(container.firstChild).toHaveStyle(`background: ${color}`);
  // });

  // it('Should change position', () => {
  //   const position = { x: 50, y: 100 };
  //   const { container } = render(
  //     <ColorSelector position={position}>
  //       <div />
  //     </ColorSelector>,
  //   );
  //   expect(container.firstChild).toHaveStyle(
  //     `left: ${position.x - 7}px, top: ${position.y}px`,
  //   );
  // });
});
