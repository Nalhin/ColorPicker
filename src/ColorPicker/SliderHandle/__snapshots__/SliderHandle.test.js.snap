// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Handle Should render correctly 1`] = `
<div
  class="css-cqd1um-SliderHandle__Wrapper e9y9zbe0"
>
  <div
    class="css-gjeaqo-SliderHandle__Handle e9y9zbe1"
    style="top: 50px; left: 33px; background: rgb(140, 150, 22);"
  />
  <div />
</div>
`;
